app/controllers/registrations_controller.rb

```diff
@@ -19,8 +20,16 @@ def create
 
   private
 
+  def redirect_if_unsupported
+    zone = Zone.find_by(zipcode: zipcode)
+
+    if zone.nil?
+      redirect_to new_zone_subscription_url(zipcode)
+    end
+  end
+
   def zone
-    Zone.find_by!(zipcode: params[:zone_id])
+    Zone.find_by!(zipcode: zipcode)
   end
```

Comment:
@geoffharcourt doesn't ActiveRecord aggressively caches the SQL for lookups?

If so, two calls to `.find_by(zipcode:)` should hit the DB once.